### PR TITLE
Welding Tool Animation

### PR DESCRIPTION
**Pull request by Tokorizo**
## About The Pull Request
Original PR: https://github.com/TauCetiStation/TauCetiClassic/pull/6540

Adds welding animations when used on walls.

## Why It's Good For The Game

A nice little animation to signal someone breaking into places, plus it's pretty well done.
102169561-8e203980-3ea3-11eb-982f-8629edee62f3

dreamseeker_2dUSJmgC4y

## Changelog

:cl:
add: Welding animation
/:cl:

### CODE REVIEW

**Review by SabreML — CHANGES_REQUESTED** [review]
Unless I'm missing something somewhere, it seems like this could be made a lot simpler and cleaner by just putting everything onto `/obj/item/weldingtool`, rather than adding things to `/obj/item/proc/use_tool()` and `/obj/item` that are only ever used by welding tools.

E.g:
```dm
/obj/item/weldingtool
	(...)
	/// An image effect for displaying a welding effect for items that support it.
	var/image/welding_sparks

/obj/item/weldingtool/Initialize(mapload)
	. = ..()
	(...)
	welding_sparks = image('code/modules/welding_sparks/icons/welding_effect.dmi', "welding_sparks", GASFIRE_LAYER)
	welding_sparks.plane = ABOVE_LIGHTING_PLANE

/obj/item/weldingtool/Destroy(force)
	QDEL_NULL(welding_sparks)
	return ..()

(...)

/obj/item/weldingtool/use_tool(atom/target, mob/living/user, delay, amount, volume, datum/callback/extra_checks)
	target.add_overlay(welding_sparks)
	. = ..()
	target.cut_overlay(welding_sparks)

File: code/game/objects/items/tools/weldingtool.dm [hunk 1]
```diff
@@ -42,12 +42,20 @@
...
 	var/deac_sound = 'sound/items/welderdeactivate.ogg'
+	/// An image effect for displaying the welding effect for items that support it
+	var/image/welding_sparks
 
 /obj/item/weldingtool/Initialize(mapload)
 	. = ..()
 	create_reagents(max_fuel)
 	reagents.add_reagent(/datum/reagent/fuel, max_fuel)
 	update_appearance()
+	welding_sparks = image('code/modules/welding_sparks/icons/welding_effect.dmi', "welding_sparks", GASFIRE_LAYER)
```
Comment (Kapu1178) [inline code/game/objects/items/tools/weldingtool.dm:53]: Icons do not belong in modules. Icons belong in `/icons`

File: code/game/objects/items/tools/weldingtool.dm [hunk 2]
```diff
@@ -42,12 +42,20 @@
 	var/burned_fuel_for = 0 //when fuel was last removed
 	var/acti_sound = 'sound/items/welderactivate.ogg'
 	var/deac_sound = 'sound/items/welderdeactivate.ogg'
+	/// An image effect for displaying the welding effect for items that support it
+	var/image/welding_sparks
```
Comment (Kapu1178) [inline code/game/objects/items/tools/weldingtool.dm:46]: Make this a static var and just set it here, doing it in initialize is a waste

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -1,5 +1,7 @@
 GLOBAL_DATUM_INIT(fire_overlay, /mutable_appearance, mutable_appearance('icons/effects/fire.dmi', "fire"))
 
+GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('code/modules/welding_sparks/icons/welding_effect.dmi', "welding_sparks", GASFIRE_LAYER, ABOVE_LIGHTING_PLANE))
```
Comment (SabreML) [inline code/game/objects/items.dm:3]: Should be fine now other than moving the dmi file into `/icons/effects`.

**Review by SabreML — APPROVED** [review]

**Review by francinum — APPROVED** [review]

**Review by axietheaxolotl — APPROVED** [review]
I am really not a fan of setting a precedent of animations being a thing. Animations work, for specific things, or things that need animation, i.e. vortex, lord scrunguloth, etc. Welding isn't something that needs an animation, should every long skill check have an animation? Should we give n-spect scanning an animation? I don't like setting a precedent of things having animations because I know from experience they are awful to do. Especially with something so common, the animation in a lone setting is good, but within the confines of everything?

That being said, it's _fine_. But if I get one person asking for an animation because "muh welding" does it, I will autistic screech the loudest I told you so.

**Pull request merged**